crates/iceberg-catalog/migrations/20240913122911_users.sql

```diff
@@ -0,0 +1,24 @@
+create type user_origin as enum ('explicit-via-register-call', 'implicit-via-config-call');
+
+create table users
```

Comment:
I think we need an additional column project_id for the search endpoint.
We need to store this information from the start, or we won't be able to add the project-scoped search endpoint later.
Same for roles I believe.

Maybe we should split project into a separate table first.

Reply:
A user entry can exist multiple times for multiple projects. Not sure if we should add another mapping table instead? It would allow us to update a single user id in all projects simultaneously. Otherwise, if we ever do support multiple IdPs, having it separated per project in the first place would make sense.